src/App.js

```diff
@@ -1,22 +1,55 @@
-import logo from './logo.svg';
 import './App.css';
+import tasks from "./data/tasks.json"
+import GenericTable from './components/table';
+import { useState } from 'react';
+import GenericModal from './components/modal/modal';
+import AddForm from './components/form/addForm';
+import EditForm from './components/form/editForm';
+import DeleteForm from './components/form/deleteForm';
 
 function App() {
+  const [data, setData] = useState(tasks);
+  const [task, setTask] = useState({});
+
+  const tableHeading = ["name", "description", "dueDate", "status"];
+
+  const addFunc = () => {
+    Object.keys(task).length === 4 && 
+    <>
+      {setData([
+        ...data,
+        task
+      ])}
+      {setTask({})}
+    </>
+      
+  }
+
+  const editFunc = () => {
+    Object.keys(task).length === 4 && 
+    <>
+      {setData(data.map(item => item.name === task.name ? task : item))}
+      {setTask({})}
+    </> 
+  }
+
+  const deleteFunc = () => {
+    Object.keys(task).length !== 0 && 
+    <>
+      {setData(data.filter(item => item.name !== task.name))}
+      {setTask({})}
+    </>
+  }
+
   return (
     <div className="App">
       <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
+        <div style={{display: 'flex'}}>
+          <GenericModal name="Add" mainForm={<AddForm task={task} setTask={setTask}/>} mainFunc={addFunc} />
```

Comment:
why are you using same modal for same work 3 times

Reply:
I am using it three times for three different buttons add, edit, and delete.